include/parse.h

```diff
@@ -53,6 +53,15 @@ netplan_parser_reset(NetplanParser *npp);
 NETPLAN_PUBLIC void
 netplan_parser_clear(NetplanParser **npp);
 
+/**
+ * @brief   Set @ref NetplanParser flags.
+ * @details Parser flags are used to change the default behavior of the parser.
+ * @param[in] npp The @ref NetplanParser to set the flags.
+ * @param[in] flags The value of the flags. The possible values are defined in @ref NETPLAN_PARSER_FLAGS
+ */
+NETPLAN_PUBLIC void
+netplan_parser_set_flags(NetplanParser* npp, int flags);
```

Comment:
suggestion: fail on unknown flags

(making the suggestion at the header level as this is API design)

I think it'd make sense to fail loudly when seeing an unknown flag rather than silently ignore it.

Reply:
+1 this should return `gboolean` and take a ` NetplanError** error` argument to be filled with details in case of failure.